src/buildstream/types.py

```diff
@@ -389,6 +389,40 @@ def new_from_node(cls, node: MappingNode) -> "_SourceMirror":
 
         return cls(name, aliases)
 
+# Used to indicate the state of a given element
+class _ElementState(FastEnum):
+    # Cannot determine the element state
+    NO_REFERENCE = "no-reference"
+
+    # The element has failed
+    FAILED = "failed"
+
+    # The element is a junction
+    JUNCTION = "junction"
+
+    # The element is waiting
+    WAITING = "waiting"
+
+    # The element is cached
+    CACHED = "cached"
+
+    # The element needs to be loaded from a remote source
+    FETCH_NEEDED = "fetch-needed"
+
+    # The element my be built
+    BUILDABLE = "buildable"
+
+    def __str__(self):
+        return str(self.value)
+
+
+# The type of encoding used when outputing machine readable information
+class _Encoding(FastEnum):
```

Comment:
Given this is a completely new command, I'm not sure we really need YAML formatting support here.

Reply:
worksforme :)